File: ruby/samples/get_forex_pairs.rb
# Load the gem
require 'intrinio-sdk'
 
# Setup authorization
Intrinio.configure do |config|
  config.api_key['api_key'] = 'YOUR_API_KEY'
end
 
forex_api = Intrinio::ForexApi.new
 
begin
  result = forex_api.get_forex_pairs
  pairs = result.pairs
  
  puts "Listing #{pairs.size} Forex Currency Pairs"
  puts ""
  pairs.each do |pair|
    puts "Code:           #{pair.code}"
    puts "Base currency:  #{pair.base_currency}"
    puts "Quote currency: #{pair.quote_currency}"
    puts "--------------------------------------------------------------------------"
  end
rescue Intrinio::ApiError => e
  puts "Exception when calling ForexApi->get_forex_pairs: #{e}"
end

File: java/JavaSDKExample/src/main/java/GetForexPairs.java
package com.intrinio.javasdkexample;

import com.intrinio.invoker.ApiClient;
import com.intrinio.invoker.ApiException;
import com.intrinio.invoker.Configuration;
import com.intrinio.invoker.auth.*;
import com.intrinio.api.ForexApi;
import com.intrinio.models.*;

import java.util.List;

public class App
{
  public static void main( String[] args )
  {
    ApiClient defaultClient = Configuration.getDefaultApiClient();
    ApiKeyAuth auth = (ApiKeyAuth) defaultClient.getAuthentication("ApiKeyAuth");
    auth.setApiKey("YOUR API KEY");

    ForexApi forexApi = new ForexApi();

    try {
      ApiResponseForexPairs result = forexApi.getForexPairs();
      List<ForexPair> pairs = result.getPairs();

      System.out.println("Listing " + pairs.size() + " Forex Currency Pairs");
      System.out.println();

      for (ForexPair pair : pairs) {
        System.out.println("Code:           " + pair.getCode());
        System.out.println("Base currency:  " + pair.getBaseCurrency());
        System.out.println("Quote currency: " + pair.getQuoteCurrency());
        System.out.println("-----------------------------------------------------------------------");
      }
    } catch (ApiException e) {
      System.err.println("Exception when calling ForexApi#getForexPairs");
      e.printStackTrace();
    }
  }
}
